torch/csrc/jit/profiling_graph_executor_impl.cpp

```diff
@@ -132,8 +132,9 @@ ExecutionPlan ProfilingGraphExecutorImpl::getPlanFor(Stack& stack) {
     return *optimized_plan_;
   }
 
+  bailout_depth_ = GetDepthAndRemoveBailOuts(graph);
```

Comment:
I don't follow this part, partially because it is mixed up in the already complicated getPlanFor logic. Why do bailout already exist in the graph at this point? I feel like the object calling the constructor to the executor can be taught to know the depth.

Reply:
> I feel like the object calling the constructor to the executor can be taught to know the depth. 
I'm putting together another version where we remember the current bailout depth in `Code` and set the new bailout depth also via `Code`. This seems to minimize the amount of threading and number of objects that need to be aware of the depth. 
I don't quite like this version either.